### PR TITLE
EVG-6687 Enable commit-queue delete command to not require passing a project

### PR DESCRIPTION
**Pull request by Chukwurado**
[EVG-6687](https://jira.mongodb.org/browse/EVG-6687)

### Description 
Make passing a project when running `command-queue delete` not required

### CODE REVIEW

File: rest/client/interface.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ type Communicator interface {
 
 	// Commit Queue
 	GetCommitQueue(ctx context.Context, projectID string) (*restmodel.APICommitQueue, error)
-	DeleteCommitQueueItem(ctx context.Context, projectID string, item string) error
```
Comment (malikchaya2) [inline rest/client/interface.go:103]: This also needs to be edited in rest/client/mock.go

**Review by malikchaya2 — COMMENTED** [review]
This looks great! I would suggest testing these changes locally or on staging to make sure that it still works as expected.

File: rest/client/methods.go [hunk 2]
```diff
@@ -1111,7 +1112,12 @@ func (c *communicatorImpl) GetCommitQueue(ctx context.Context, projectID string)
 	return &cq, nil
 }
 
-func (c *communicatorImpl) DeleteCommitQueueItem(ctx context.Context, projectID, item string) error {
+func (c *communicatorImpl) DeleteCommitQueueItem(ctx context.Context, item string) error {
+	requestedPatch, err := patch.FindOneId(item)
```
Comment (ablack12) [inline rest/client/methods.go:1116]: need to check if requestedPatch == nil

**Review by ablack12 — CHANGES_REQUESTED** [review]
Need to increment client version string https://github.com/evergreen-ci/evergreen/blob/cc9bd2c636945353e6748be842e441e83b58a616/config.go#L35

File: rest/client/methods.go [hunk 3]
```diff
@@ -1111,7 +1112,15 @@ func (c *communicatorImpl) GetCommitQueue(ctx context.Context, projectID string)
...
+	if err != nil {
+		return errors.Wrap(err, "error finding item")
+	}
+	if requestedPatch == nil {
+		return errors.Wrap(err, "item not found")
```
Comment (hadjri) [inline rest/client/methods.go:1121]: Since error is nil here you can just return an error with the message only.

File: config.go [hunk 4]
```diff
@@ -32,7 +32,7 @@ var (
 	BuildRevision = ""
 
 	// Commandline Version String; used to control auto-updating.
-	ClientVersion = "2021-12-17"
+	ClientVersion = "2021-12-22"
```
Comment (hadjri) [inline config.go:35]: Need to increment one more time, unfortunately its updated again since we started review.

**Review by hadjri — APPROVED** [review]
LGTM, go ahead and merge once my 2 comments are addressed!

**Pull request merged**